### ISSUE TITLE
Best Practices for IP-Adapter-FaceID

### ISSUE THREAD

**Issue by whiterose199187** [tencent-ailab/IP-Adapter#197]
Hello,

I am trying out IP-Adapter-FaceID

1) What is recommendation for input image size? 
2) What is recommendation for scale/guidance_scale?

Thanks

**Comment by xiaohu2015** [comment]
hi, it should work for any image size, you can try adjust any parameters to get best results.

**Comment by gurteshwar** [comment]
ok, thanks

**Comment by whiterose199187** [comment]
ok thanks

**Comment by blx0102** [comment]
With the same model, the output of faceid version is relative darker than ori face version, what's the possible reason?

**Comment by xiaohu2015** [comment]
> With the same model, the output of faceid version is relative darker than ori face version, what's the possible reason?

ori face version is ?

**Comment by blx0102** [comment]
@xiaohu2015

**Comment by blx0102** [comment]
@xiaohu2015 
Let me make it more clear. Below is an example, left two columns were generated by using "ip-adapter-plus-face_sd15.bin", while right two columns were generated by using "ip-adapter-faceid-plus_sd15.bin". 
The right two are a little darker than left two, both for the girl's shin and the environment light. They all used the same prompt. 
![image](https://github.com/tencent-ailab/IP-Adapter/assets/17761044/9046b6b3-37b5-4d63-9e34-06993361f2af)

**Comment by xiaohu2015** [comment]
The two models are different in design, so they may have different results.

**Comment by cubiq** [comment]
I believe there's some bias coming from the previous models. All IPAdapter models "burn" the image quite a bit so you generally need to lower the weight, the CFG scale (and generally increase the steps).

The new FaceID models are far less aggressive and they actually act more "naturally" as they should. This also means that you can use a very strong weight (hence increase the face likeliness) and use very aggressive CFG scale value.

With faceID you can safely use weight 1.2 and CFG 9 and you will get more or less the same saturation you were getting from a Full Face with the added benefit that you can reach good results with fewer steps.

This is an example both at 20 steps (white plus face would probably require 30 at least)

The first picture is FaceID with CFG 6 and weight 0.8
![ComfyUI_temp_epjur_00020_](https://github.com/tencent-ailab/IP-Adapter/assets/427614/b5b77d51-83c6-457b-a61c-2e33e91b5eeb)

This second one if FaceID with CFG 9 and weight 9.0
![ComfyUI_temp_epjur_00021_](https://github.com/tencent-ailab/IP-Adapter/assets/427614/a68770c3-be7d-4dfa-a5ff-de4f68bcc97a)

As you can see you can reach the same kind of saturation you had before with Plus Face.

TL;DR: it's FacePlus that is wrong not FaceID :smile:

**Comment by xiaohu2015** [comment]
> I believe there's some bias coming from the previous models. All IPAdapter models "burn" the image quite a bit so you generally need to lower the weight, the CFG scale (and generally increase the steps).
> 
> The new FaceID models are far less aggressive and they actually act more "naturally" as they should. This also means that you can use a very strong weight (hence increase the face likeliness) and use very aggressive CFG scale value.
> 
> With faceID you can safely use weight 1.2 and CFG 9 and you will get more or less the same saturation you were getting from a Full Face with the added benefit that you can reach good results with fewer steps.
> 
> This is an example both at 20 steps (white plus face would probably require 30 at least)
> 
> The first picture is FaceID Plus with CFG 6 and weight 0.8 ![ComfyUI_temp_epjur_00020_](https://private-user-images.githubusercontent.com/427614/293329191-b5b77d51-83c6-457b-a61c-2e33e91b5eeb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM4Mzk1NDEsIm5iZiI6MTcwMzgzOTI0MSwicGF0aCI6Ii80Mjc2MTQvMjkzMzI5MTkxLWI1Yjc3ZDUxLTgzYzYtNDU3Yi1hNjFjLTJlMzNlOTFiNWVlYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyOVQwODQwNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yOTI3NzU4OTY1OWQzZmI5OTM2ZWU2MjI0MTczYzg5YmVkYzViZjU5NWNlMmU4YmNkMTA0MmI3ODg0NGZmYWEwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hq6KdfZBUvJL-dKjiaGMwjOsxgtPGGwKM4EmJexeR1I)
> 
> This second one if FaceID Plus with CFG 9 and weight 1.2 ![ComfyUI_temp_epjur_00021_](https://private-user-images.githubusercontent.com/427614/293329187-a68770c3-be7d-4dfa-a5ff-de4f68bcc97a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM4Mzk1NDEsIm5iZiI6MTcwMzgzOTI0MSwicGF0aCI6Ii80Mjc2MTQvMjkzMzI5MTg3LWE2ODc3MGMzLWJlN2QtNGRmYS1hNWZmLWRlNGY2OGJjYzk3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyOVQwODQwNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jM2M1ODdkMGQzZjI1MjRmM2U4MjM4Njc3YWM1MmZiNDc3OWU2Nzk2NmQwOWUyZTVhNTM5YWI2ZmQxNzliOWMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GlDsqb3GNFFKeygmeE_1WsK1yxefSYKT7yW9LLnMN_E)
> 
> As you can see you can reach the same kind of saturation you had before with Plus Face.
> 
> TL;DR: it's FacePlus that is wrong not FaceID 😄

in fact, we use some different training tricks to train FaceID Plus, so it is more robust. FaceID Plus v2 will support adjust the weight of face structure.

by the way, the result you show is FaceID Plus, how about FaceID?

**Comment by cubiq** [comment]
FaceID is different because it really needs to be used together with PlusFace or FullFace.

When used alone it doesn't "burn" the image as much as Plus/Full Face but more than FaceIDPlus

This is weight 0.8 CFG 6
![ComfyUI_temp_epjur_00025_](https://github.com/tencent-ailab/IP-Adapter/assets/427614/186ae009-f51e-4700-929f-c592d963e183)

This is weight 1.2 CFG 9
![ComfyUI_temp_epjur_00026_](https://github.com/tencent-ailab/IP-Adapter/assets/427614/16b30454-063a-450f-be89-c6b2e83ec963)
